src/acquisition/flusurv/flusurv.py

```diff
@@ -142,78 +173,192 @@ def mmwrid_to_epiweek(mmwrid):
+    #     [
+    #         {'networkid': 1, 'catchmentid': 22, 'seasonid': 49, 'ageid': 0, 'sexid': 0, 'raceid': 1, 'rate': 4.3, 'weeklyrate': 1.7, 'mmwrid': 2493},
+    #         {'networkid': 1, 'catchmentid': 22, 'seasonid': 49, 'ageid': 0, 'sexid': 0, 'raceid': 1, 'rate': 20.3, 'weeklyrate': 0.1, 'mmwrid': 2513},
+    #         {'networkid': 1, 'catchmentid': 22, 'seasonid': 49, 'ageid': 0, 'sexid': 0, 'raceid': 1, 'rate': 20.6, 'weeklyrate': 0.1, 'mmwrid': 2516},
+    #         ...
+    #     ]
+    for obs in data:
+        epiweek = mmwrid_to_epiweek(obs["mmwrid"])
+        season = id_season_map[obs["seasonid"]]
+        groupname = groupids_to_name(
+            ageid = obs["ageid"], sexid = obs["sexid"], raceid = obs["raceid"],
+            id_label_map = id_label_map
+        )
+
+        # Set season description. This will be overwritten every iteration,
+        #  but should always have the same value per epiweek group.
+        data_out[epiweek]["season"] = season
+
+        rate = obs["weeklyrate"]
+        prev_rate = data_out[epiweek][groupname]
+        if prev_rate is None:
+            # This is the first time to see a rate for this epiweek-group
+            #  combo
+            data_out[epiweek][groupname] = rate
+        elif prev_rate != rate:
+            # Skip and warn; a different rate was already found for this
+            # epiweek-group combo
+            warn((f"warning: Multiple rates seen for {epiweek} "
+                   f"{groupname}, but previous value {prev_rate} does not "
+                   f"equal new value {rate}. Using the first value."))
+
+    # Sanity check the input. We expect to have populated our dictionary
+    if len(data_out.keys()) == 0:
+        raise Exception("no data loaded")
+
+    print(f"found data for {len(data_out.keys())} epiweeks")
+
     return data_out
 
 
-def get_data(location_code):
+def get_data(location, seasonids, metadata):
     """
     Fetch and parse flu data for the given location.
 
     This method performs the following operations:
-      - fetches FluSurv data from CDC
-      - extracts and returns hospitaliation rates
+      - fetch location-specific FluSurv data from CDC API
+      - extracts and returns hospitalization rates for each epiweek
     """
-
     # fetch
     print("[fetching flusurv data...]")
-    data_in = fetch_flusurv_object(location_code)
+    data_in = fetch_flusurv_location(location, seasonids)
 
     # extract
-    print("[extracting values...]")
-    data_out = extract_from_object(data_in)
+    print("[reformatting flusurv result...]")
+    data_out = group_by_epiweek(data_in, metadata)
 
     # return
-    print("[scraped successfully]")
+    print(f"[successfully fetched data for {location}]")
     return data_out
 
 
-def get_current_issue():
-    """Scrape the current issue from the FluSurv main page."""
-
-    # fetch
-    data = fetch_json("GetPhase03InitApp?appVersion=Public", None)
+def get_current_issue(data):
```

Comment:
this is only called on the response from `fetch_flusurv_metadata()`, right?
```suggestion
def get_current_issue(metadata):
```